pkg/inventory/types.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2021 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package inventory
+
+import "k8s.io/apimachinery/pkg/types"
+
+// Inventory represents the inventory object in memory.
+// Inventory is currently only used for in-memory storage and not serialized to
+// disk or to the API server.
+// TODO: Replace InventoryInfo with Inventory.TypeMeta & Inventory.ObjectMeta
+// TODO: Replace object.ObjMetadataSet in Storage interface with Inventory.Spec
+type Inventory struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   InventorySpec   `json:"spec,omitempty"`
+	Status InventoryStatus `json:"status,omitempty"`
+}
+
+// InventorySpec is the specification of the desired/expected inventory state.
+type InventorySpec struct {
+	Objects []ObjectReference `json:"objects,omitempty"`
+}
+
+// InventoryStatus is the status of the current/last-known inventory state.
+type InventoryStatus struct {
+	Objects []ObjectStatus `json:"objects,omitempty"`
+}
+
+// ObjectReference is a reference to a KRM resource by name and kind.
+//
+// Kubernetes only stores one API Version for each Kind at any given time,
+// so version is not used when referencing objects.
+type ObjectReference struct {
+	// Kind identifies a REST resource within a Group.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	Kind string `json:"kind,omitempty"`
+
+	// Group identifies an API namespace for REST resources.
+	// If group is omitted, it is treated as the "core" group.
+	// More info: https://kubernetes.io/docs/reference/using-api/#api-groups
+	// +optional
+	Group string `json:"group,omitempty"`
+
+	// Name identifies an object instance of a REST resource.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	Name string `json:"name,omitempty"`
+
+	// Namespace identifies a group of objects across REST resources.
+	// If namespace is specified, the resource must be namespace-scoped.
+	// If namespace is omitted, the resource must be cluster-scoped.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// ObjectStatus is a snapshot of the actuation and reconciliation status of a
+// referenced object.
+type ObjectStatus struct {
+	ObjectReference `json:",inline"`
+
+	Strategy  ActuationStrategy `json:"strategy,omitempty"`
+	Actuation ActuationStatus   `json:"actuation,omitempty"`
+	Reconcile ReconcileStatus   `json:"reconcile,omitempty"`
```

Comment:
Please add some comments for these fields.

Reply:
Done.